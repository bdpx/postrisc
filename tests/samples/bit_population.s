.text
    alloc	25
    ldi.l	%r23, 0x1234567890abcdef
    write	"test population statistic instructions"
    cnt.pop	%r12, %r23, 3
    write	"cntpop: %i64(r12)"
    cnt.lz	%r12, %r23, 0
    write	"cntlz %i64(r12)"
    cnt.tz	%r12, %r23, 1
    cnt.lz	%r12, %r23, 2
    cnt.tz	%r12, %r23, 3
    cnt.lz	%r12, %r23, 4
    cnt.tz	%r12, %r23, 5
.end
