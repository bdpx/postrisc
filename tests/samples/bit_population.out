test population statistic instructions
cntpop: 4
cntlz 0
